### DISCUSSION TITLE
Improvement of datetime type caster for very distant dates.

### DISCUSSION THREAD

**Discussion by jpvergnes** [wjakob/nanobind#400] [Ideas]
It seems that the current implementation of type caster for datetime.datetime object doesn't allow very distant dates to be transfered (e.g. 2400-01-01). I need to transfer such date with seconds as Duration. In my case, a workaround was to replace line 167 in nanobind/stl/chrono.h : 

`value = ch::time_point_cast<Duration>(
             ch::system_clock::from_time_t(std::mktime(&cal)) + msecs);`

by

`value = type(Duration(std::mktime(&cal)));`

and my super distant date are well transferred from python to c++. Would it be feasible to introduce such a feature in a next release ?

**Comment by wjakob** [comment]
That seems like it would cause other issues (loss of millisecond-precision). Perhaps the numeric precision needs to be changed -- cc @oremanj
